From 6f0e2f65b3711fe8e9c814f849166e3a6356a3c0 Mon Sep 17 00:00:00 2001
From: Franck Jullien <franck.jullien@gmail.com>
Date: Tue, 10 Mar 2020 11:43:48 +0100
Subject: [PATCH] scripts/bareboximd: add mode argument to open call

Newer gcc complain about missing mode argument when a new file is
created. This is an extract of man-pages:

 "The mode argument specifies the file mode bits be applied when
 a new file is created.  This argument must be supplied when
 O_CREAT or O_TMPFILE is specified in flags"

Signed-off-by: Franck Jullien <franck.jullien@gmail.com>
Signed-off-by: Sascha Hauer <s.hauer@pengutronix.de>
---
 scripts/bareboximd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/bareboximd.c b/scripts/bareboximd.c
index e5000e0ae..b332d435a 100644
--- a/scripts/bareboximd.c
+++ b/scripts/bareboximd.c
@@ -59,7 +59,7 @@ static int write_file(const char *filename, const void *buf, size_t size)
 	int fd, ret;
 	int now;
 
-	fd = open(filename, O_WRONLY | O_TRUNC | O_CREAT);
+	fd = open(filename, O_WRONLY | O_TRUNC | O_CREAT, S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH);
 	if (fd < 0)
 		return fd;
 
-- 
2.17.1

